### ISSUE TITLE
Choosing which features to calculate in settings

### ISSUE THREAD

**Issue by matus-tomlein** [blue-yonder/tsfresh#153]
I have the following case:

1. I extract features from a training dataset
2. I filter the extracted features
3. I want to extract the same kinds of features as were filtered in step 2 from the test dataset

So far, I haven't been able to figure out how to make tsfresh calculate only specific features based on their keys. I can extract all features in step 3 and then filter them so that they only contain ones with the same keys as the ones from training dataset, but that means that a lot of features are calculated unnecessarily and it is slowing down the testing time.

I wasn't able to find any option in the extraction settings that would do this.
Is there any way this could be done?

Thanks for the awesome library!

**Comment by MaxBenChrist** [comment]
Hi @matus-tomlein , indeed we have to improve the documentation with respect to this functionality (especially about the `from_colums` method)

You can use the sklearn compatible RelevantFeatureAugmenter http://tsfresh.readthedocs.io/en/latest/text/sklearn_transformers.html. After fitting it, it will only extract features that are relevant and thus prevent uncessary calculations.

**Comment by matus-tomlein** [comment]
Thanks, that's useful!

Maybe I'm missing something, but is the pipeline approach also applicable in case I am classifying data not present in the dataset that I originally fitted the RelevantFeatureAugmenter with?
What I mean is a workflow like:

1. I have a dataset of pre-collected sensor values that I used to find the relevant features
2. Later I receive new data from a sensor deployed in the field
3. I only extract the relevant features I found in step 1 from this new data (and classify them)

It seems to me that the pipeline is designed for a more offline system, because you use the indices in the data you send to the fit() function that refer to the data you set in the params of the augmenter.
I'm not sure if that fits a real-time system that continuously receives new data.
Or am I missing something?

**Comment by MaxBenChrist** [comment]
> It seems to me that the pipeline is designed for a more offline system, because you use the indices in the data you send to the fit() function that refer to the data you set in the params of the augmenter.
I'm not sure if that fits a real-time system that continuously receives new data.
Or am I missing something?

I don't understand that paragraph 

From what I getting you describe a situation with a train and a test set. For this use case the RelevantFeatureAugmentor was developed

**Comment by MaxBenChrist** [comment]
If you are not familiar with sklearn pipelines, you can look up the sklearn documentation

**Comment by MaxBenChrist** [comment]
A short example

You are developing your pipeline
```python
pipeline.set_params(augmenter__timeseries_container=df_old)
pipeline.fit(X_old, y_old)
```
Now the pipeline object contains the settings of which features to calculate.

If you now call

```python
pipeline.set_params(augmenter__timeseries_container=df_new)
pipeline.transform(X_new)
```
it will avoid uncessary calculations and only calculate those features that were useful for `df_old`

**Comment by MaxBenChrist** [comment]
Besides those runs, you can save your settings by just pickling the fitted augmentar object

**Comment by matus-tomlein** [comment]
Thanks!

I was confused because I didn't realize that is possible to call `pipeline.set_params(augmenter__timeseries_container=df)` twice without having to run pipeline.fit() again.
Now it makes sense.

**Comment by MaxBenChrist** [comment]
No Problem, Happy it helped you :). You could also consider to support us by improving the wording of the documentation or submitting a short notebook that demonstrates how to use the augmenter.
